CRM/ManualDirectDebit/Mail/Task/ContributionEmailCommon.php

```diff
@@ -109,33 +109,44 @@ public static function submit(&$form, $formValues) {
       $contributionIds = $form->getVar('_contributionIds');
     }
 
+    $errors = [];
     foreach ($formattedContactDetails as $formattedContactDetail) {
       $contactId = $formattedContactDetail['contact_id'];
       $selectedContributionIdsForCurrentContact = self::getSelectedContributionIdsForCurrentContact($contactId, $contributionIds);
       foreach ($selectedContributionIdsForCurrentContact as $contributionId) {
-        $dataCollector = new CRM_ManualDirectDebit_Mail_DataCollector_Contribution($contributionId);
-        $tplParams = $dataCollector->retrieve();
-        $contactDetail = [$formattedContactDetail];
-        CRM_ManualDirectDebit_Mail_Task_Mail::sendEmail(
-          $contactDetail,
-          $subject,
-          $formValues['text_message'],
-          $formValues['html_message'],
-          NULL,
-          NULL,
-          $from,
-          $attachments,
-          $cc,
-          $bcc,
-          array_keys($form->_toContactDetails),
-          $additionalDetails,
-          $contributionIds,
-          CRM_Utils_Array::value('campaign_id', $formValues),
-          $tplParams
-        );
+        try {
```

Comment:
maybe move everything under the 2nd foreach into a new method (e.g sendEmail() ) to make it easier to read a little.